### ISSUE TITLE
Redis Sentinel

### ISSUE THREAD

**Issue by VagyokC4** [thepirat000/Audit.NET#493]
Are we able to use Redis Sentinel to take advantage of failovers?

**Comment by thepirat000** [comment]
Please add more context. Otherwise the response is obviously Yes, you can

**Comment by VagyokC4** [comment]
Forgive my ignorance.  I'm looking to test our solution using a Redis Sentinel cluster.  Ok, so I guess the question then is how do I wire up a redis sentinel connection?

I am currently using the StackExchange.Redis.Extensions.Core.Configuration.RedisConfiguration which allows me to specify hosts, as well as a sentinel name.

When I get down the audit.net, it seems I can only provide a connection string.  How would I configure audit.net to use a sentinel cluster to manage which redis is the master?

**Comment by thepirat000** [comment]
Oh, I see. 

I don't know the library you mentioned, but what I can do is make the changes to allow providing the `ConfigurationOptions` instead of the connection string. Will that help?

**Comment by VagyokC4** [comment]
> Oh, I see.
> 
> I don't know the library you mentioned, but what I can do is make the changes to allow providing the [`ConfigurationOptions`](https://stackexchange.github.io/StackExchange.Redis/Configuration.html) instead of the connection string. Will that help?

Yes, because that gives me the "serviceName" property that I can use to wire up sentinel.

**Comment by thepirat000** [comment]
This was implemented on version 19.1.3, please upgrade your references and re-test

Now you can pass the `ConfigurationOptions` instance, for example:

```c#
Audit.Core.Configuration.Setup()
    .UseRedis(redis => redis
        .ConfigurationOptions(configurationOptions)
        .AsList(list => list
            .Key("AuditEvents")
            .MaxLength(1000)));
```
